### DISCUSSION TITLE
Clever and ClassLink authentication

### DISCUSSION THREAD

**Discussion by NikilXYZ** [supabase/supabase#15032] [Feature Requests]
I work with school districts, and they often expect apps to have a "log in with Clever" or "log in with Classlink" social authentication feature. I believe they both use OAuth.

Can you add those?

They also should have an ability to import student rosters using OneRoster. 

How difficult is that to add?

Thanks!

**Comment by sjohns21** [comment]
i'd also like this

https://dev.clever.com/docs/oauth-implementation

i explored the SAML solution
but supabase SAML requires an email for every user
and Clever does not gurantee an email from their users

@thorwebdev

**Comment by hf** [comment]
Hey, we're working on adding generic OAuth / OIDC support for any identity provider as well as getting rid of the email requirement. Stay tuned in 2024!

**Reply by burstWizard** [reply]
Hi, are there any updates on this?

**Reply by brycedbjork** [reply]
@hf when can we expect this?

**Comment by brycedbjork** [comment]
Any update on this?

**Comment by jacobgoldfarb** [comment]
Would also love this!

**Comment by nigeltroy-gpt0** [comment]
Would greatly appreciate this! 🙌
